Language filter initial draft

rzhao271 opened the pull request:
Ref https://github.com/microsoft/vscode/issues/131707

## Summary

This PR adds an initial draft of the language filter:
- `@lang:` is suggested in the search widget
- `@lang:langId` shows all language-overridable settings, but the values shown correspond to either the default setting value, or the language-specific setting value. TODO: determine whether we are showing/can show the language-specific overridden default value whenever it shows up.
- When a language filter has multiple languages set, we only read the first one for now to determine which language to filter by
- When a setting is modified while a language filter is set, the setting is modified as a language-specific setting value for the given language
- When a setting is modified while a language filter is set (even when outside of the workspace scope), the setting remains in the Settings editor JSON file until the user explicitly resets it. This change makes it easier to debug the feature, but I'm not sure how necessary it is.

## Demo

![Language filter demo screencap](https://user-images.githubusercontent.com/7199958/158274636-486840c7-b11f-42ae-8fc3-84f792b08a62.gif)

roblourens commented on src/vs/workbench/contrib/preferences/browser/settingsEditor2.ts:
```diff
@@ -948,21 +955,22 @@ export class SettingsEditor2 extends EditorPane {
 		const resource = URI.isUri(settingsTarget) ? settingsTarget : undefined;
 		const configurationTarget = <ConfigurationTarget>(resource ? ConfigurationTarget.WORKSPACE_FOLDER : settingsTarget);
-		const overrides: IConfigurationOverrides = { resource };
+		const overrides: IConfigurationUpdateOverrides = { resource, overrideIdentifiers: languageFilters };
```
Does this set the setting in multiple languages? It seems more likely that someone would do that by accident, especially if we only look at the first one for displaying the filtered settings.

rzhao271 replied:
I changed the filters field from type `string[]` or `Set<string>` to type `string | undefined`, and changed the variable names to be singular rather than plural. This way, it could be more clear that only one language is being read for now.

roblourens commented:
It would be helpful if I could get suggestions for language IDs after typing `@lang:`. I can't necessarily remember whether what the ID of a langage is, and I get this in settings.json.

Also, if I use a language ID that doesn't exist, I still get results, and can't tell that anything is wrong.

roblourens commented:
I have one setting in a [html] block. When I enter `@lang:html @modified`, I see 27 modified settings. I actually don't know what all of these are?

rzhao271 commented on src/vs/platform/configuration/common/configurationRegistry.ts:
```diff
@@ -9,6 +9,7 @@ import { Emitter, Event } from 'vs/base/common/event';
 import { IJSONSchema } from 'vs/base/common/jsonSchema';
 import * as types from 'vs/base/common/types';
 import * as nls from 'vs/nls';
+import { getLanguageTagSettingPlainKey } from 'vs/platform/configuration/common/configuration';
```
Test comment.

rzhao271 commented:
There were a few bugs with deciding what to render when that I fixed above.
I plan on waiting for the CI build, and then building an Insiders that you can test with, first @roblourens.

rzhao271 commented:
Fixed another bug with the `@modified` tag.
I also added a constant to make it so that it's easier to enable or disable this feature, in case testing doesn't go well.
